staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset_test.go

```diff
@@ -993,6 +993,79 @@ func TestSelectQueueLocked(t *testing.T) {
 	}
 }
 
+func TestFinishRequestLocked(t *testing.T) {
+	tests := []struct {
+		name         string
+		workEstimate fcrequest.WorkEstimate
+	}{
+		{
+			name: "request has additional latency",
+			workEstimate: fcrequest.WorkEstimate{
+				Seats:             10,
+				AdditionalLatency: time.Minute,
+			},
+		},
+		{
+			name: "request has no additional latency",
+			workEstimate: fcrequest.WorkEstimate{
+				Seats: 10,
+			},
+		},
+	}
+
+	metrics.Register()
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			metrics.Reset()
+
+			now := time.Now()
+			clk, _ := testeventclock.NewFake(now, 0, nil)
+			qs := &queueSet{
+				clock:   clk,
+				obsPair: newObserverPair(clk),
+			}
+			queue := &queue{
+				requests: newRequestFIFO(),
+			}
+			r := &request{
+				qs:           qs,
+				queue:        queue,
+				workEstimate: test.workEstimate,
+			}
+
+			qs.totRequestsExecuting = 111
+			qs.totSeatsInUse = 222
+			queue.requestsExecuting = 11
+			queue.seatsInUse = 22
+
+			var (
+				queuesetTotalRequestsExecutingExpected = qs.totRequestsExecuting - 1
+				queuesetTotalSeatsInUseExpected        = qs.totSeatsInUse - int(test.workEstimate.Seats)
+				queueRequestsExecutingExpected         = queue.requestsExecuting - 1
+				queueSeatsInUseExpected                = queue.seatsInUse - int(test.workEstimate.Seats)
+			)
+
+			qs.finishRequestLocked(r)
+
+			// as soon as AdditionalLatency elapses we expect the seats to be released
```

Comment:
If we really want to lock down the behavior of `finishRequestLocked`, we could also check the values one nanosecond before the proper release time.

Reply:
Yes, I will update the test to assert on this.